### PR TITLE
Bump phpunit, add psalm, dump app:code-check

### PR DESCRIPTION
**Pull request by nickvergessen**

### CODE REVIEW

File: tests/psalm-baseline.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <NullArgument occurrences="1">
+      <code>null</code>
+    </NullArgument>
+  </file>
+  <file src="lib/Dashboard/Widget.php">
+    <InvalidArgument occurrences="1"/>
+  </file>
+  <file src="lib/Model/Announcement.php">
+    <UndefinedMethod occurrences="2">
+      <code>parent::getMessage()</code>
```
Comment (ChristophWurst) [inline tests/psalm-baseline.xml:13]: why is it not `$this->getMessage()`? I think Psalm is right here.

Comment (nickvergessen) [reply tests/psalm-baseline.xml:13]: For message this works, or is the same. On getSubject() I didn't use a different name in my model.
And the problem is that `OCP\AppFramework\Db\Entity` works with `__call` and therefore yes Psalm is right, the method does not exist, but it works :-X

Comment (ChristophWurst) [reply tests/psalm-baseline.xml:13]: I see. Carry on :speak_no_evil:

**Review by ChristophWurst — APPROVED** [review]

File: tests/psalm-baseline.xml (hunk 1 above)
Comment (nickvergessen) [reply tests/psalm-baseline.xml]: Fixed

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**